frontend/src/metabase/public/containers/PublicOrEmbeddedDashboard/WithPublicDashboardEndpoints.tsx

```diff
@@ -7,26 +7,26 @@ import {
 } from "metabase/services";
 import type { DashboardId } from "metabase-types/api";
 
+/** @deprecated - prefer `usePublicDashboardEndpoints`*/
 export const WithPublicDashboardEndpoints = <T extends WithRouterProps>(
   Component: ComponentType<T>,
 ): ComponentType<T & { dashboardId: DashboardId }> => {
-  function DashboardEndpointsInner({
-    params: { uuid, token },
-    ...props
-  }: {
-    params: {
-      uuid?: string;
-      token?: string;
-    };
-  }) {
-    if (uuid) {
-      setPublicDashboardEndpoints();
-    } else if (token) {
-      setEmbedDashboardEndpoints();
-    }
-    const dashboardId = uuid || token;
+  function DashboardEndpointsInner(props: WithRouterProps) {
+    const { dashboardId } = usePublicDashboardEndpoints(props);
     return <Component {...(props as T)} dashboardId={dashboardId} />;
   }
 
   return DashboardEndpointsInner;
 };
+
+export const usePublicDashboardEndpoints = (props: WithRouterProps) => {
+  const { uuid, token } = props.params;
+  if (uuid) {
+    setPublicDashboardEndpoints();
+  } else if (token) {
+    setEmbedDashboardEndpoints();
+  }
```

Comment:
do we need a useEffect here as this is technically a side effect? looks like these resolves to setting `DashboardApi.parameterValues` in the end though, so I suppose it's not required as the function is idempotent.